test/integration/connection-test.js

```diff
@@ -359,6 +359,17 @@ describe('Encrypt Test', function() {
   });
 });
 
+describe('BeginTransaction Tests', function() {
+  it('should throw invalid IsolationLevel Error', function() {
+    const config = getConfig();
+    config.options.isolationLevel = 9;
+    const connection = new Connection(config);
```

Comment:
You probably should wait for the connection to be established before you call `connection.beginTransaction`. 🤔